pkg/cli/ignition/convert3.go

```diff
@@ -0,0 +1,116 @@
+package ignition
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+
+	"github.com/coreos/ign-converter/translate/v24tov31"
+	ignv2 "github.com/coreos/ignition/config/v2_4"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	kcmdutil "k8s.io/kubectl/pkg/cmd/util"
+	"k8s.io/kubectl/pkg/util/templates"
+)
+
+const convert3CommandName = "convert3"
+
+type convert3Options struct {
+	// Source file
+	Source string
+	// Destination file
+	Destination string
+
+	SourceReader io.Reader
+	DestWriter   io.Writer
```

Comment:
if the options struct becomes public, all the fields of the struct will also become public API, maybe we can make all of these private for start with or only leave certain fields public that we think will be public API when this struct becomes public...

Reply:
Yeah, will fix this though I'll note that this pattern seems fairly old and pervasive in `oc`; I think I used [extract.go](https://github.com/openshift/oc/blob/8150994e0392adb4aeade70d62974ca77b3740a4/pkg/cli/extract/extract.go) as a template.